### ISSUE TITLE
Markdown highlighter doesn't work with nested code block in list item

### ISSUE THREAD

**Issue by leira** [mawww/kakoune#3582]
### Steps 
1. Edit a simple .md file in kakoune with content:
~~~markdown
* 1st item
* ```
  funcall()
  ```
* 3rd item

## new header
~~~

### Outcome 
The highlighter doesn't recognize the start of a code block after `* `, but take the ending "```" as the start of a code block, thus messed up all the highlighting. Like this:

![Screenshot 2020-07-06 at 5 03 09 PM](https://user-images.githubusercontent.com/39118/86673311-664fb000-bfac-11ea-8e54-b34a47b37b88.png)

### Expected 
Expect the highlighter to recognize a code blocker nested in a list item.

**Comment by Carsenal** [comment]
After having poked around a little bit, this seems kinda tough. According to the common mark syntax, while your example should be a code block, the following should *not* be considered a code block (ignore the spaces between backticks, I couldn't get it to render right):
```
* 1st item
* ` ` `
funcall()
` ` `
```
Related, [this test codeblock](https://markdown-it.github.io/#md3=%7B%22source%22%3A%22%23%20Lists%20baby%2C%5Cn%2A%20Item%201%5Cn%2A%20Item%202%5Cn%2A%20%60%60%60%5Cn%20%20item%203%20is%20code%5Cn%20%20%20%60%60%60%5Cn%2A%20Item%204%5Cn%2A%20%60%60%60%5CnItem%205%20%2Ashould%2A%20be%20code%5Cn%60%60%60%5Cnbut%20it%20misrenders%22%2C%22defaults%22%3A%7B%22html%22%3Afalse%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22html%22%7D%7D)
Because of the nature of markdown indentation, I think the current implementation of highlighters might be unfit to process whitespace in this way. It is possible I am mistaken, but I suspect we may not be able to fully conform to spec here without changes.

**Comment by lenormf** [comment]
This issue is fixed by #3582.
